packages/editor/src/editor/browser/cms/useTextSelection.ts

```diff
@@ -0,0 +1,58 @@
+import { useState } from 'react';
+
+export type Selection = { start: number; end: number };
+export type InputTextAreaRef = React.RefObject<HTMLInputElement> | React.RefObject<HTMLTextAreaElement>;
+
+const useTextSelection = (ref: InputTextAreaRef) => {
+  const [selection, setSelection] = useState<Selection | undefined>(undefined);
+  const handleTextSelection = () => {
+    if (ref.current) {
+      const selectionStart = ref.current.selectionStart ?? 0;
+      const selectionEnd = ref.current.selectionEnd ?? 0;
+      if (selectionStart !== selectionEnd) {
+        setSelection({ start: selectionStart, end: selectionEnd });
+      } else {
+        setSelection(undefined);
+      }
+    }
+  };
+
+  const getSelectedText = () => {
+    if (ref.current && selection) {
+      const text = ref.current.value.substring(selection.start, selection.end);
+      return text;
+    }
+    return '';
+  };
+
+  const showQuickFix = () => {
+    if (!selection || !ref.current) return false;
```

Comment:
I know sometimes this single line `if`'s makes sense
But they are not very common in our code bases, and not everyone of our team likes them. 
So I would reduce the usage of them to a minium and more write 
```tsx
if () {
  return false;
}
```

Reply:
All right, i can replace this with the normal if 👍. I thought this might be an elegant solution so that the code is not too bloated.